libmamba/support-shared-libs.patch

```diff
@@ -1,8 +1,8 @@
 diff --git a/libmamba/CMakeLists.txt b/libmamba/CMakeLists.txt
```

Comment:
Could you please provide some context on why this patch is needed? Under what circumstances can we safely remove this patch? how can we recreate the patch if required? 
these details will help ensure the patch is maintainable.: https://eng.inky.wtf/docs/squads/sustaining/playbooks/patch-creation-guide/

Reply:
I agree this is something we should be doing in general, but as you can see this PR only refreshes the patch for the newer version. @dannf might not have the required context to determine what the patch was about. I think this should be a separate PR if anything.
We shouldn't block on adding missing information that was missing in the package in the first place.

Reply:
+1 - patch originated from here https://github.com/wolfi-dev/os/pull/29763#issuecomment-2463427683 , I just rebased it. But I 100% agree with the feedback. I added this link into the patch file for lack of more context.